feat: Add dependency vulnerability scans to supported tool table TAROT-2398 (#1896)
codacy/docs: docs/getting-started/supported-languages-and-tools.md

      <td><a href="https://github.com/SonarSource/sonar-dotnet">SonarC#</a></td>
      <td>-</td>
      <td><a href="https://trivy.dev">Trivy</a></td>

>       <td><a href="https://trivy.dev">Trivy</a> (scans <code>packages.lock.json</code>, <code>packages.config</code>, <code>.deps.json</code>) <a href="#any-path"><sup>3</sup></a></td><!-- TODO TAROT-2398 confirm that these file formats match what we analyze -->

Author: Does Trivy scan all these?

Reviewer: Sorry, it's best to look at this table, 'Filesystem' column -- https://aquasecurity.github.io/trivy/v0.45/docs/coverage/language/#supported-languages

          <a href="https://pmd.github.io/">PMD</a></td>
      <td><a href="https://eslint.org/docs/rules/">ESLint</a> <a href="#suggest-fixes">🔧</a></td>
      <td><a href="https://trivy.dev">Trivy</a></td>

>       <td><a href="https://trivy.dev">Trivy</a> (scans <br><code>package-lock.json</code>, <br><code>yarn.lock</code>) <a href="#any-path"><sup>3</sup></a></td>

Reviewer 2: are these enough, or would it be also required to add the "package.json"? 🤔

Author: I'll mention the tool name as well. I'd keep the file name since that's what's actually scanned; here I'm assuming that it may be useful for the user to have a written confirmation that a specific file is/isn't being scanned.

      <td>-</td>
      <td>-</td>
      <td>-</td>
      <td>-</td>
    </tr>
    <tr>
      <td>Python</td>
      <td><a href="https://github.com/PyCQA/bandit">Bandit</a>,
          <a href="https://github.com/landscapeio/prospector">Prospector</a>,
          <a href="https://github.com/pylint-dev/pylint">Pylint</a></td>
      <td>-</td>
      <td><a href="https://github.com/PyCQA/bandit">Bandit</a>,
          <a href="https://github.com/landscapeio/prospector">Prospector</a>,
          <a href="https://trivy.dev">Trivy</a></td>
      <td><a href="https://trivy.dev">Trivy</a> (scans <code>Pipfile.lock</code>) <a href="#any-path"><sup>3</sup></a></td>
      <td><a href="https://pmd.github.io/pmd/pmd_userdocs_cpd.html">PMD CPD</a></td>
      <td><a href="https://github.com/rubik/radon">Radon</a></td>
    </tr>
    <tr>
      <td>Ruby<a href="#ruby-31"><sup>6</sup></a></td>
      <td><a href="https://brakemanscanner.org/">Brakeman</a>,
          <a href="https://github.com/rubocop/rubocop">RuboCop</a></td>
      <td>-</td>
      <td><a href="https://trivy.dev">Trivy</a></td>

>       <td><a href="https://trivy.dev">Trivy</a> (scans <code>gemfile.lock</code>) <a href="#any-path"><sup>3</sup></a></td>

Reviewer 2: ```suggestion
      <td><a href="https://trivy.dev">Trivy</a> (scans <code>Gemfile.lock</code>) <a href="#any-path"><sup>3</sup></a></td>
```

      <td>-</td>
      <td>-</td>
      <td>-</td>
      <td>-</td>
    </tr>
    <tr>
      <td>Swift </td>
      <td><a href="https://github.com/realm/SwiftLint">SwiftLint</a>,
          <a href="https://github.com/sleekbyte/tailor">Tailor</a></td>
      <td>-</td>
      <td>-</td>

>       <td><a href="https://trivy.dev">Trivy</a> (scans <code>Package.resolved</code>) <a href="#any-path"><sup>3</sup></a></td><!-- TODO TAROT-2398 Do we also scan Cocoa pods (Podfile.lock)? -->

Reviewer 3: I don't think so

      <td><a href="https://trivy.dev">Trivy</a></td>
      <td>-</td>
      <td>-</td>
      <td>-</td>
    </tr>
  </tbody>
</table>

<sup><span id="yaml-only">1</span></sup>: Currently, Trivy only supports scanning YAML files on this platform.  
<sup><span id="client-side">2</span></sup>: Supported as a [client-side tool](../repositories-configure/local-analysis/client-side-tools.md).  

> <sup><span id="any-path">3</span></sup>: The path of these files does not matter.

Reviewer 4: > The path of these files does not matter.

@nicklem What does this mean? 🤔 

Also, check the formatting. It's not breaking correctly to (4).

Author: Good catch @claudiacarpinteiro! Here's a reworded version, let me know if it's clearer:

```suggestion
<sup><span id="any-path">3</span></sup>: These files will be scanned regardless of their file path.  
```

> <sup><span id="any-path">3</span></sup>: These files are scanned regardless of their file path.  

Reviewer: Not sure what you mean by this but the current logic is to let Trivy analyse all files but only report issues with line numbers for files that Codacy told Trivy to analyse.

For example, if the dependencies file is ignored on Codacy, no issues will be reported for it.

Author: On second thought, this info is not vital. I'll remove it for now and reevaluate if we get any complaints.
